manager/volume.go

```diff
@@ -17,24 +19,70 @@ import (
 	longhorn "github.com/rancher/longhorn-manager/k8s/pkg/apis/longhorn/v1alpha1"
 )
 
+type BundleState string
+
+const (
+	BundleStateInProgress  = BundleState("InProgress")
+	BundleReadyForDownload = BundleState("ReadyForDownload")
+	BundleStateError       = BundleState("Error")
+)
+
+type ErrorString string
+
+const (
+	MkdirFailed = ErrorString("Failed to create bundle file directory")
+	ZipFailed   = ErrorString("Failed to compress the support bundle files")
+	OpenFailed  = ErrorString("Failed to open the compressed bundle file")
+	StatFailed  = ErrorString("Failed to compute the size of the compressed bundle file")
+	None        = ErrorString("none")
+)
+
+type SupportBundle struct {
+	State      BundleState
+	Size       int64
+	File       io.ReadCloser
+	Error      ErrorString
+	Fname      string
+	createTime time.Time
+}
+
+func NewSupportBundle() *SupportBundle {
```

Comment:
Don't need this. See above.

Reply:
We still need a way to create a SupportBundle object. Why do you think we don't need this constructor?

Reply:
It's the default constructor. You can just get it by `sb := SupportBundle{}`